src/compat.h

```diff
@@ -16,9 +16,24 @@
 #include <pgstat.h>
 #include <utils/lsyscache.h>
 #include <utils/rel.h>
+#include <access/xact.h>
 
 #include "export.h"
 
+/* This is a copy of PreventCommandIfReadOnly() where we only do the function
+ * call if the transaction is a read-only. If we were to use the function
+ * directly, we would evaluate the function name regardless of whether we were
+ * in a read-only transaction or not */
+#define TS_ERROR_IF_READ_ONLY()                                                                    \
```

Comment:
I think the macro name should somehow connect to the fact that this macro can only be used in a SQL-invoked function context. Now it looks like a general macro that can be used in other contexts as well. For instance:

```
TS_PREVENT_FUNC_IF_READ_ONLY()
```

Reply:
Renamed.